package com.iod.pb.test;

import "common/IODCommon.proto";

message ReqAuthentication {
	optional string user_id = 1;
	optional string authentication = 2;
}

message ResAuthentication {
	optional sint32 result = 1;	
	optional string authorization = 2;
}

message ReqLogin {
  optional string user_id = 1;
  optional string authorization = 2;
}

message ResLogin {
  optional string user_id = 1;
  optional sint32 result = 2;	
}

message ReqTestInfo {
  optional string info = 1;
}

message ResTestInfo {
  optional string info = 1;
}

message ReqLogout {
}

message ReqTestResponseTime {
	required sint64 req_timestamp = 1;
}

message ResTestResponseTime {
	required sint64 req_timestamp = 1;
}

message NotifyKickout {
	optional sint32 kick_reason = 1;
}

message TestMsg1 {
	optional bytes test_data = 1;
}
message TestMsg2 {
	optional bytes test_data = 1;
}
message TestMsg3 {
	optional bytes test_data = 1;
}
message TestMsg4 {
	optional bytes test_data = 1;
}
message TestMsg5 {
	optional bytes test_data = 1;
}
message TestMsg6 {
	optional bytes test_data = 1;
}
message TestMsg7 {
	optional bytes test_data = 1;
}
message TestMsg8 {
	optional bytes test_data = 1;
}
message TestMsg9 {
	optional bytes test_data = 1;
}
message TestMsg10 {
	optional bytes test_data = 1;
}
message TestMsg11 {
	optional bytes test_data = 1;
}
message TestMsg12 {
	optional bytes test_data = 1;
}
message TestMsg13 {
	optional bytes test_data = 1;
}
message TestMsg14 {
	optional bytes test_data = 1;
}
message TestMsg15 {
	optional bytes test_data = 1;
}
message TestMsg16 {
	optional bytes test_data = 1;
}
message TestMsg17 {
	optional bytes test_data = 1;
}
message TestMsg18 {
	optional bytes test_data = 1;
}
message TestMsg19 {
	optional bytes test_data = 1;
}
message TestMsg20 {
	optional bytes test_data = 1;
}

extend com.iod.pb.common.BaseMsg {
	optional ReqAuthentication idReqAuthentication = 1000;
	optional ResAuthentication idResAuthentication = 1001;
	optional ReqLogin idReqLogin = 1002;
	optional ResLogin idResLogin = 1003;
	optional ReqTestInfo idReqTestInfo = 1004;
	optional ResTestInfo idResTestInfo = 1005;
	optional ReqLogout idReqLogout = 1006;	
	optional ReqTestResponseTime idReqTestResponseTime = 1007;
	optional ResTestResponseTime idResTestResponseTime = 1008;
	optional NotifyKickout idNotifyKickout = 1009;
	optional TestMsg1 idTestMsg1 =5001;
	optional TestMsg2 idTestMsg2 =5002;
	optional TestMsg3 idTestMsg3 =5003;
	optional TestMsg4 idTestMsg4 =5004;
	optional TestMsg5 idTestMsg5 =5005;
	optional TestMsg6 idTestMsg6 =5006;
	optional TestMsg7 idTestMsg7 =5007;
	optional TestMsg8 idTestMsg8 =5008;
	optional TestMsg9 idTestMsg9 =5009;
	optional TestMsg10 idTestMsg10 =5010;
	optional TestMsg11 idTestMsg11 =5011;
	optional TestMsg12 idTestMsg12 =5012;
	optional TestMsg13 idTestMsg13 =5013;
	optional TestMsg14 idTestMsg14 =5014;
	optional TestMsg15 idTestMsg15 =5015;
	optional TestMsg16 idTestMsg16 =5016;
	optional TestMsg17 idTestMsg17 =5017;
	optional TestMsg18 idTestMsg18 =5018;
	optional TestMsg19 idTestMsg19 =5019;
	optional TestMsg20 idTestMsg20 =5020;
}